### PR TITLE
chore (refs T34082): add missing Hooks/Selectors & E2E Props.

### PR DESCRIPTION
**Pull request by ahmad-demos**
https://yaits.demos-deutschland.de/T34082
https://github.com/demos-europe/demosplan-ui/pull/443

- add missing Hooks/Selectors & E2E Props.

### CODE REVIEW

File: client/js/components/procedure/basicSettings/AutoSwitchProcedurePhaseForm.vue [hunk 1]
```diff
@@ -119,6 +123,16 @@ export default {
...
 
+    dataCyEndDate: {
+      type: String,
+      required: false
+    },
+
+    dataCyStartDate: {
+      type: String,
+      required: false
+    },
```
Comment (salisdemos) [inline client/js/components/procedure/basicSettings/AutoSwitchProcedurePhaseForm.vue:134]: for not required fields , we always should use a default

**Review by salisdemos — CHANGES_REQUESTED** [review]

File: client/js/components/procedure/basicSettings/AutoSwitchProcedurePhaseForm.vue (hunk 1 above)
Comment (ahmad-demos) [reply client/js/components/procedure/basicSettings/AutoSwitchProcedurePhaseForm.vue]: Done in https://github.com/demos-europe/demosplan-core/pull/1860/commits/3599e2edea3302d1b997804176e852f77f2b4ef6

File: client/js/components/procedure/basicSettings/AutoSwitchProcedurePhaseForm.vue [hunk 2]
```diff
@@ -119,6 +123,18 @@ export default {
       default: () => ({})
     },
 
+    dataCyEndDate: {
+      type: String,
+      required: false,
+      default: ''
```
Comment (salisdemos) [inline client/js/components/procedure/basicSettings/AutoSwitchProcedurePhaseForm.vue:129]: shouldn't the default be something maningfull like "endDate" ?

File: client/js/components/procedure/basicSettings/AutoSwitchProcedurePhaseForm.vue [hunk 3]
```diff
@@ -119,6 +123,18 @@ export default {
...
+    },
+
+    dataCyStartDate: {
+      type: String,
+      required: false,
+      default: ''
```
Comment (salisdemos) [inline client/js/components/procedure/basicSettings/AutoSwitchProcedurePhaseForm.vue:135]: see above

**Review by salisdemos — CHANGES_REQUESTED** [review]

File: client/js/components/procedure/basicSettings/AutoSwitchProcedurePhaseForm.vue (hunk 2 above)
Comment (ahmad-demos) [reply client/js/components/procedure/basicSettings/AutoSwitchProcedurePhaseForm.vue]: Done in https://github.com/demos-europe/demosplan-core/pull/1860/commits/3a5154131d8f0ea2008098faa275812a8b2d2a87, thnx

File: client/js/components/procedure/basicSettings/AutoSwitchProcedurePhaseForm.vue (hunk 3 above)
Comment (ahmad-demos) [reply client/js/components/procedure/basicSettings/AutoSwitchProcedurePhaseForm.vue]: Done in https://github.com/demos-europe/demosplan-core/pull/1860/commits/3a5154131d8f0ea2008098faa275812a8b2d2a87, thnx

**Review by salisdemos — APPROVED** [review]

**Pull request merged**